main/lsp/LSPIndexer.cc

```diff
@@ -299,14 +299,12 @@ LSPFileUpdates LSPIndexer::commitEdit(SorbetWorkspaceEditParams &edit) {
         ENFORCE(runningSlowPath.epoch <= pendingTypecheckUpdates.epoch);
         ENFORCE(runningSlowPath.epoch > (pendingTypecheckUpdates.epoch - pendingTypecheckUpdates.editCount));
 
-        auto merged = update.copy();
-        merged.mergeOlder(pendingTypecheckUpdates);
-        merged.canTakeFastPath = canTakeFastPath(merged, true);
-        // Cancel if old + new takes fast path, or if the new update will take the slow path anyway.
-        if ((merged.canTakeFastPath || !update.canTakeFastPath) &&
-            initialGS->epochManager->tryCancelSlowPath(merged.epoch)) {
+        // Cancel if the new update will take the slow path anyway.
+        if (!update.canTakeFastPath && initialGS->epochManager->tryCancelSlowPath(update.epoch)) {
             // Cancelation succeeded! Use `merged` as the update.
```

Comment:
Trivial nit:  there is no `merged` variable after this change, but yeah, you still use a merged update, so maybe just drop the quotes (or reword, if you like):
```suggestion
            // Cancelation succeeded! Use merged as the update.
```